contracts/PinningManager.sol

```diff
@@ -209,66 +197,31 @@ contract PinningManager {
             numberOfPeriodsDeposited <= MAX_UINT64,
             "PinningManager: period too long"
         );
+        //NO_OVERFLOW reasoning: verified above
+        now.add(numberOfPeriodsDeposited * period); // overFlow check. If this doesn't pass, the duration of the offer overflows MAX_UINT64 and the contract may deadlock. REF_DURATION
         numberOfPeriodsDeposited.mul(chosenPrice); // overFlow check. If this doesn't pass, we might have issues transfering the maximum amount REF_MAX_TRANSFER
         request.chosenPrice = chosenPrice;
         request.chosenPeriod = period;
```

Comment:
I believe here should be `offerRegistry[provider].periods[period]` as the `period` should be array's index, right?

Reply:
See previous comment about period. I think this question can be resolved. Let me know if you think otherwise

Reply:
I still believe it stands. See my previous comment.